src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -880,13 +573,676 @@ def defineFaceAnnotations(cls, region, options, annotationGroups):
+        d2Along = []
+        for n2 in range(len(domeNodes_x)):
+            xAlong.append(domeNodes_x[n2][n1])
+            d2Along.append(domeNodes_d2[n2][n1])
+        xSampledAlongDome, d2SampledAlongDome = interp.sampleCubicHermiteCurves(xAlong, d2Along,
+                                                                                elementsCountAlongDome,
+                                                                                arcLengthDerivatives=True)[0:2]
+        d2Smoothed = interp.smoothCubicHermiteDerivativesLine(xSampledAlongDome, d2SampledAlongDome)
+        xRaw.append(xSampledAlongDome)
+        d2Raw.append(d2Smoothed)
+
+    # Rearrange x and d2
+    xSampledDome = []
+    d1SampledDome = []
+    d2SampledDome = []
+    for n2 in range(elementsCountAlongDome + 1):
+        xAround = []
+        d1Around = []
+        d2Around = []
+        for n1 in range(elementsCountAround):
+            x = xRaw[n1][n2]
+            d2 = d2Raw[n1][n2]
+            xAround.append(x)
+            d2Around.append(d2)
+            # Calculate d1
+            if n2 > 0:
+                v1 = xRaw[n1][n2]
+                v2 = xRaw[n1 + 1 if n1 < elementsCountAround - 2 else 0][n2]
+                d1 = findDerivativeBetweenPoints(v1, v2)
+                d1Around.append(d1)
+            else:
+                d1Around.append(d2Raw[n2][0])
+        if n2 > 0:
+            d1Smoothed = interp.smoothCubicHermiteDerivativesLoop(xAround, d1Around)
+        else:
+            d1Smoothed = d1Around
+        xSampledDome.append(xAround)
+        d1SampledDome.append(d1Smoothed)
+        d2SampledDome.append(d2Around)
+
+    return xSampledDome, d1SampledDome, d2SampledDome
+
+def findNodesAlongBladderNeck(sx_dome_group, sx_neck_group, domeSegmentLength, neckSegmentLength, elementsCountAround, elementsCountAlongNeck):
+
+    # Transition
+    transitLength = (domeSegmentLength + neckSegmentLength) / 2
+    addingLength = transitLength - neckSegmentLength
+    if transitLength <= neckSegmentLength:
+        e = 1
+        xi = transitLength / neckSegmentLength
+    else:
+        e = int(transitLength / neckSegmentLength) + 1
+        xi = transitLength / (e * neckSegmentLength)
+    xTransition = interp.interpolateCubicHermite(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    # d1Transition = interp.interpolateCubicHermiteDerivative(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    d2Transition = interp.interpolateCubicHermite(sx_dome_group[2][-1], sx_dome_group[3][-1], sx_neck_group[2][e], sx_neck_group[3][e], xi)
+    d3Transition = interp.interpolateCubicHermite(sx_dome_group[4][-1], sx_dome_group[5][-1], sx_neck_group[4][e], sx_neck_group[5][e], xi)
+    px_transit, pd1_transit = createEllipsePoints(xTransition, 2 * math.pi, d2Transition, d3Transition, elementsCountAround,
+                                                  startRadians=0.0)
```

Comment:
I haven't looked into this in depth, but would think the proper use of sampling along and smoothing along the next would work fine as described below.

Reply:
I can remove lines 794-805 since I only used "addingLength" later in the "sampleCubicHermiteCurves" functions when I created the neck part. Could you please have a look to this function and "findNodesAlongBladderNeck" function. It seems something is wrong here that makes the surface derivatives not being smooth.